.changeset/pink-ligers-dance.md

```diff
@@ -0,0 +1,18 @@
+---
+'@leafygreen-ui/icon': minor
+---
+
+Updates `createIconComponent` to handle both number and string values passed to the `size` prop on custom glyphs. 
+
+e.g.
+```jsx
+  import { createIconComponent, Size} from '@leafygreen-ui/icon'
+
+  const myIcon = ({size}) => <svg height={size} width={size}> ... </svg>
```

Comment:
My initial concern is still valid, but this is still a useful feature to add! We should consider a more versatile solution in the future